cns/deviceplugin/server.go

```diff
@@ -0,0 +1,157 @@
+package deviceplugin
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1"
+)
+
+const devicePrefix = "NIC-"
+
+type deviceCounter interface {
+	getDeviceCount() int
+}
+
+type Server struct {
+	address                  string
+	logger                   *zap.Logger
+	deviceCounter            deviceCounter
+	listAndWatchShutdownChan chan struct{}
```

Comment:
I think we should just call it `shutdownCh` since the point is to communicate to shut down the server

Reply:
updated